include/CppUTest/CppUTestConfig.h

```diff
@@ -281,7 +281,7 @@
 #endif
 #endif
 
-#if CPPUTEST_USE_LONG_LONG
+#ifdef CPPUTEST_USE_LONG_LONG
```

Comment:
Could we keep this as an #if. In general we prefer #if over #ifdef. In this case, it would likely require a change at the previous #if block to include an #else ?

Reply:
All other checks are `#ifdef`s right now which is the cause of this regression. I can change them all if that's what you prefer.

Reply:
All other checks on the CPPUTEST_USE (which can be used outside the Config file) are always done with  #if. That one is set based on different conditions. So, in this case, the block before this one (which checks whether CPPUTEST_USE_LONG_LONG should be set to 1) probably ought to change so that it will set it to 0 is the case no long long is used. That way it will be consistent with the rest of the file.

Reply:
I made the change. I ran into trouble with GCC's '-Wuninitialized` and with DOS trying to get `cpputest_longlong` and `cpputest_ulonglong` handled. I can't figure out how they're useful, so I removed them.

Reply:
The reason is that we cannot have #if in the header files to change the v table structure. It leads to very very unexpected errors.

Reply:
I'm not sure I follow. That would only happen if different values of `CPPUTEST_USE_LONG_LONG` were defined for different translation units, which would be a problem no matter what.

Anyway, I've rolled back my removal of `cpputest_longlong` and `cpputest_ulonglong`. They break the build.

Reply:
I think I've found a compromise. See my latest commit. There's no way to set up `cpputest_longlong` and `cpuutest_ulonglong` so that assignment works _and_ C linkage works. I managed to retain ABI stability. The only interface change required is to disable the `LONGLONGS_EQUAL`, `UNSIGNED_LONGLONGS_EQUAL`, etc. macros when `CPPUTEST_USE_LONG_LONG` is unset.

Reply:
Thanks, I'll have a look at it. It has happened in the past that the CPPUTEST_USE_LONG_LONG was set for different translation units (set differently when compiling CppUTest itself than for the tests). The big problem with it is that it doesn't give an error but behaves very surprising :)

As a result, we avoid #if in headers and especially for virtual functions.